framework/decode/file_processor.cpp

```diff
@@ -494,27 +494,46 @@ bool FileProcessor::ProcessBlocks()
 // the correct sizing of the block payload are done by the caller
 bool FileProcessor::ReadBlockBuffer(BlockParser& parser, BlockBuffer& block_buffer)
 {
-    bool success = parser.ReadBlockBuffer(GetCurrentFile().active_file, block_buffer);
-    if (success)
+    bool           success = true;
+    BlockReadError status  = parser.ReadBlockBuffer(GetCurrentFile().active_file, block_buffer);
+    if (status == kErrorNone)
```

Comment:
I feel like this flow is difficult to follow.  I imagine this could be simplified into just the case `(status != kErrorNone)` which returns `false` and otherwise returns `true` with no `success` but I don't want to block this PR.  So simplify this at the next opportunity?